pkg/nextgen/distdoc/graph_test.go

```diff
@@ -0,0 +1,175 @@
+package distdoc
+
+import (
+	"bytes"
+	"crypto"
+	"crypto/x509"
+	"encoding/binary"
+	"github.com/nuts-foundation/nuts-network/pkg/model"
+	"github.com/stretchr/testify/assert"
+	"strings"
+	"testing"
+)
+
+// trackingVisitor just keeps track of which nodes were visited in what order.
+type trackingVisitor struct {
+	documents []Document
+}
+
+func (n *trackingVisitor) Accept(document Document, next []Document) {
+	n.documents = append(n.documents, document)
+}
+
+func (n trackingVisitor) JoinContentsAsString() string {
+	var contents []string
+	for _, document := range n.documents {
+		buf := new(bytes.Buffer)
+		buf.ReadFrom(document.Payload())
+		contents = append(contents, buf.String())
+	}
+	return strings.Join(contents, ", ")
+}
+
+func TestChain_Add(t *testing.T) {
+	key, certificate := generateKeyAndCertificate()
+	t.Run("ok", func(t *testing.T) {
+		graph := NewDAG()
```

Comment:
very compact tests.

Would it be an idea to add 2 newlines so we can distinguish a given/when/then approach or setup/test/assert?